beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -556,40 +551,38 @@ func (s *Server) PrepareBeaconProposer(w http.ResponseWriter, r *http.Request) {
 		httputil.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
 		return
 	}
-	var feeRecipients []common.Address
 	var validatorIndices []primitives.ValidatorIndex
 	// filter for found fee recipients
 	for _, r := range jsonFeeRecipients {
 		validatorIndex, valid := shared.ValidateUint(w, "validator_index", r.ValidatorIndex)
 		if !valid {
-			return
+			log.WithField("validatorIndex", validatorIndex).Warn("fee recipient is invalid")
+			continue
 		}
 		feeRecipientBytes, valid := shared.ValidateHex(w, "fee_recipient", r.FeeRecipient, fieldparams.FeeRecipientLength)
 		if !valid {
-			return
+			log.WithField("validatorIndex", validatorIndex).Warn("fee recipient is invalid")
+			continue
 		}
-		f, err := s.BeaconDB.FeeRecipientByValidatorID(ctx, primitives.ValidatorIndex(validatorIndex))
-		switch {
-		case errors.Is(err, kv.ErrNotFoundFeeRecipient):
-			feeRecipients = append(feeRecipients, common.BytesToAddress(bytesutil.SafeCopyBytes(feeRecipientBytes)))
-			validatorIndices = append(validatorIndices, primitives.ValidatorIndex(validatorIndex))
-		case err != nil:
-			httputil.HandleError(w, fmt.Sprintf("Could not get fee recipient by validator index: %v", err), http.StatusInternalServerError)
-			return
-		default:
-			if common.BytesToAddress(feeRecipientBytes) != f {
-				feeRecipients = append(feeRecipients, common.BytesToAddress(bytesutil.SafeCopyBytes(feeRecipientBytes)))
-				validatorIndices = append(validatorIndices, primitives.ValidatorIndex(validatorIndex))
+		// Use default address if the burn address is return
+		feeRecipient := primitives.ExecutionAddress(feeRecipientBytes)
+		if feeRecipient == primitives.ExecutionAddress([20]byte{}) {
```

Comment:
We also have a constant that represents this in hex format without having to do a conversion.